js/tests/test_examples.test.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * E2E Tests for plugin demos and examples.
+ */
+
+const { execFile } = require('child_process');
+const path = require('path');
+const { promisify } = require('util');
+const execFileAsync = promisify(execFile);
+
+// Paths
+const PLUGINS = path.resolve(__dirname, '../plugins');
+const EXAMPLES_PATH = path.resolve(__dirname, '../examples');
+
+// Require env vars
+const { OPENAI_API_KEY, COMPOSIO_API_KEY, JULEP_API_KEY, JULEP_API_URL, LISTENNOTES_API_KEY } = process.env;
+console.log(COMPOSIO_API_KEY);
+
+// Plugin test definitions
+const EXAMPLES = {
+  // autogen: {
+  //   plugin: 'autogen',
+  //   file: path.join(PLUGINS, 'autogen', 'autogen_demo.js'),
+  //   match: {
+  //     type: 'stdout',
+  //     values: ['Action executed successfully'],
+  //   },
+  //   env: {
+  //     OPENAI_API_KEY,
+  //     COMPOSIO_API_KEY,
+  //   },
+  // },
+  // llamaindex: {
+  //   plugin: 'llamaindex',
+  //   file: path.join(PLUGINS, 'llamaindex', 'llamaindex_demo.js'),
+  //   match: {
+  //     type: 'stdout',
+  //     values: ['Action executed successfully'],
+  //   },
+  //   env: {
+  //     OPENAI_API_KEY,
+  //     COMPOSIO_API_KEY,
+  //   },
+  // },
+  // local_tools: {
+  //   plugin: 'autogen',
+  //   file: path.join(EXAMPLES_PATH, 'miscellaneous', 'math_agent', 'autogen_math.js'),
+  //   match: {
+  //     type: 'stdout',
+  //     values: ['11962'],
+  //   },
+  //   env: { OPENAI_API_KEY },
+  // },
+  runtime_tools: {
+    plugin: 'langchain',
+    file: path.join(EXAMPLES_PATH, 'scheduling-agent', 'demo.mjs'),
+    match: {
+      type: 'stdout',
+      values: ['🎉Output from agent: '],
+    },
+    env: { COMPOSIO_API_KEY },
+  },
+  // crew_ai: {
+  //   plugin: 'crew_ai',
+  //   file: path.join(PLUGINS, 'crew_ai', 'crewai_demo.js'),
+  //   match: {
+  //     type: 'stdout',
+  //     values: ['Action executed successfully'],
+  //   },
+  //   env: {
+  //     OPENAI_API_KEY,
+  //     COMPOSIO_API_KEY,
+  //   },
+  // },
+  // Add more examples as needed
+};
+
+describe('E2E Tests for plugin demos and examples', () => {
+  jest.setTimeout(180000); // Set timeout to 3 minutes
+
+  for (const [exampleName, example] of Object.entries(EXAMPLES)) {
+    test(`should run ${exampleName} example successfully`, async () => {
+      const pluginToTest = process.env.PLUGIN_TO_TEST;
+      if (pluginToTest && pluginToTest !== example.plugin) {
+        return test.skip();
+      }
+
+      for (const [key, val] of Object.entries(example.env)) {
+        expect(val).not.toBeNull();
+      }
+      console.log(example.env);
```

Comment:
Another instance of potentially unsafe logging. Consider removing this or replacing with a safer logging format that doesn't expose sensitive environment variables.